docs/_templates/autoapi/index.rst

```diff
@@ -0,0 +1,13 @@
+Code Documentation
+==================
+
+This page contains reference documentation generated from code comments [#f1]_.
+
+.. toctree::
+   :titlesonly:
+
+   {% for page in pages|selectattr("is_top_level_object") %}
+   {{ page.include_path }}
+   {% endfor %}
+
+.. [#f1] Created with `sphinx-autoapi <https://github.com/readthedocs/sphinx-autoapi>`_
```

Comment:
In order to customize the title of this section and to remove the "Submodules" nesting, I copy-pasted all templates from the "autoapi" project here and used them directly. We can edit them further if/when we have to.

It would have been nice to only take the one template we needed, but I think we need all of them.